src/Client/Resources/keen-io-3_0.php

```diff
@@ -1247,6 +1247,146 @@
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+            'additionalParameters' => array(
+                'location' => 'json'
+            ),
+        ),
+        'listAccessKeys' => array(
+            'uri'         => 'projects/{projectId}/keys',
+            'description' => 'Returns all project access keys.',
+            'httpMethod'  => 'GET',
+            'parameters'  => array(
+                'projectId'        => array(
+                    'location'    => 'uri',
+                    'type'        => 'string'
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+        ),
+        'getAccessKey' => array(
+            'uri'         => 'projects/{projectId}/keys/{key}',
+            'description' => 'Returns a project access key.',
+            'httpMethod'  => 'GET',
+            'parameters'  => array(
+                'projectId'        => array(
+                    'location'    => 'uri',
+                    'type'        => 'string'
+                ),
+                'key'      => array(
+                    'location'    => 'uri',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+        ),
+        'updateAccessKey' => array(
+            'uri'         => 'projects/{projectId}/keys/{key}',
+            'description' => 'Updates a project access key.',
+            'httpMethod'  => 'POST',
+            'parameters'  => array(
+                'projectId'        => array(
+                    'location'    => 'uri',
+                    'type'        => 'string'
+                ),
+                'key'      => array(
+                    'location'    => 'uri',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
```

Comment:
This is really similar to creating an access key. I wonder if there's a nice way to not have to redefine the parameters that they share so all of that isn't repeated? Maybe by defining the creation parameters elsewhere and merging that array here with the key parameter?